### ISSUE TITLE
Support reopened action on PullRequestEvent

### ISSUE THREAD

**Issue by bradleyfalzon** [bradleyfalzon/gopherci#25]
Reopened occurs when a closed PR is reopened, the CI appears to run again, so should we.

https://github.com/bradleyfalzon/gopherci/blob/37c3f1aa887d5ac5c7867d9a108f0853924bcfa4/internal/github/handlers.go#L76-L78

Related #24

**Comment by bradleyfalzon** [comment]
I'm not convinced we should, if the hash doesn't change, why should we re-run checks?

**Comment by shurcooL** [comment]
It would make sense to rerun if your check depends on the base branch. For example, a PR might've been opened and closed with a change which was not an API breaking change. But later, that PR gets reopened and by then, the same change becomes a breaking one.

If your check doesn't depend on base branch, nothing should change if a PR is simply reopened.

**Comment by bradleyfalzon** [comment]
`apicompat` does check the base branch. But I can't think of a case where the base branch updating would cause `apicompat` to detect an issue - but it could potentially clear an issue.

But this wouldn't catch the case if the base branch was updated, but the PR didn't change, I'd still need to separately handle that.

And as people are billed per build, gopherci.io wouldn't want to execute more builds than necessary. But this is gopherci.io's problem, I'd rather they just give free rebuilds on reopen/base changing, than having billing get in the way of a correct solution.

So perhaps the case is:

- If a PR is open, and the base branch changes, rebuild.
- If the PR is re-opened and the base branch has changed, rebuild.

Is that sounding correct?

**Comment by shurcooL** [comment]
Sounds reasonable to me. Plus you can always iterate on this later, if it becomes clear there's a better behavior available.

**Comment by bradleyfalzon** [comment]
This has been resolved, I've opted to just rebuild on every reopen, we can optimise this later.

I won't bother opening an issue for `If a PR is open, and the base branch changes, rebuild. ` only apicompat depends on the base branch, and it's not clear whether there's value in rerunning these checks (unless anyone can point out changes to a base branch causing the head to suddenly have a breaking change).

There's potentially value in rerunning tests, or perhaps if it can merge, merge and run the tests again - there may be some value there.

But for the moment, because it's non-trivial, I'll leave this closed.
